lib/Site.js

```diff
@@ -204,6 +205,47 @@ Site.prototype.buildSourceFiles = function () {
   });
 };
 
+/**
+ * Rebuild rebuild source files that are affected by change in filePath
+ * @param filePath path of file changed
+ */
+Site.prototype.rebuildSourceFiles = function (filePath) {
+  return new Promise((resolve, reject) => {
+    this.generateRelevantPages(filePath)
+      .then(() => fs.removeAsync(this.tempPath))
+      .then(resolve)
+      .catch((error) => {
+        // if error, remove the site and temp folders
+        rejectHandler(reject, error, [this.tempPath, this.outputPath]);
+      });
+  });
+};
+
+/**
+ * Generates pages that contain the original file path
+ * as the source file or as static/dynamic included file
+ * @param filePath path of file changed
+ */
+Site.prototype.generateRelevantPages = function (filePath) {
+  const builtFiles = {};
+  const processingFiles = [];
+  this.pageModels.forEach((page) => {
+    if (page.includedFiles[filePath]) {
+      processingFiles.push(page.generate(builtFiles));
+    }
+  });
+
+  logger.info(`Rebuilding ${processingFiles.length} pages`);
+
+  return new Promise((resolve, reject) => {
+    Promise.all(processingFiles)
+      .then(() => logger.info('Files Rebuilt'))
+      .then(resolve)
+      .catch(reject);
+  });
+};
```

Comment:
- Move `generateRelevantPages` after `generatePages`.
- Should it be called `regenerateAffectedPages` since:
  - it only `re`generates pages that were previously generated (and consistent with `rebuildSourceFiles`), and
  - it's not obvious what's considered `Relevant`? (`Affected` is marginally better.)